### PR TITLE
support ReactTestUtils in decorators that use forwardRef

### PR DESCRIPTION
**Pull request by balzss**
INSTUI-3893

### CODE REVIEW

File: packages/emotion/src/__tests__/withStyle.test.tsx [hunk 1]
```diff
@@ -136,6 +138,28 @@ describe('@withStyle', async () => {
...
+    render() {
+      return (
+        <div>
+          <ThemeableComponent />
+        </div>
+      )
+    }
+  }
+
+  it.only('can be found and tested with ReactTestUtils', async () => {
```
Comment (matyasf) [inline packages/emotion/src/__tests__/withStyle.test.tsx:151]: remove `.only`

File: packages/emotion/src/__tests__/withStyle.test.tsx [hunk 2]
```diff
@@ -136,6 +138,28 @@ describe('@withStyle', async () => {
...
+    const rootNode = document.createElement('div')
+    document.body.appendChild(rootNode)
+
+    // eslint-disable-next-line react/no-render-return-value
+    const rendered = ReactDOM.render(<WrapperComponent />, rootNode)
+    ReactTestUtils.findRenderedComponentWithType(
+      rendered as any,
+      (ThemeableComponent as any).originalType
+    )
```
Comment (matyasf) [inline packages/emotion/src/__tests__/withStyle.test.tsx:160]: This test has no assertion. Or will it throw an error if its not found?

**Review by matyasf — CHANGES_REQUESTED** [review]

File: packages/emotion/src/__tests__/withStyle.test.tsx (hunk 2 above)
Comment (balzss) [reply packages/emotion/src/__tests__/withStyle.test.tsx:160]: it will and then the test fails

**Review by joyenjoyer — APPROVED** [review]
nice work

File: packages/emotion/src/__tests__/withStyle.test.tsx (hunk 1 above)
Comment (balzss) [reply packages/emotion/src/__tests__/withStyle.test.tsx]: done

**Review by matyasf — APPROVED** [review]

**Pull request merged**